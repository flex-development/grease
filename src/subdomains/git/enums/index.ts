/**
 * @file Entry Point - Enums
 * @module grease/git/enums
 */

export { default as RepositoryProvider } from './repository-provider'
