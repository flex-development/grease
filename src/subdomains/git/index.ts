/**
 * @file Entry Point - git
 * @module grease/git
 */

export * from './enums'
export { default as GitModule } from './git.module'
export type * from './interfaces'
export * from './models'
export * from './options'
export * from './providers'
export type * from './types'
