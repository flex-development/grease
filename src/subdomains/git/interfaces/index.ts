/**
 * @file Entry Point - Interfaces
 * @module grease/git/interfaces
 */

export type { default as ICommit } from './commit.interface'
