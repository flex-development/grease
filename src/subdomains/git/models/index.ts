/**
 * @file Entry Point - Models
 * @module grease/git/models
 */

export { default as CommitGrammar } from './commit-grammar.model'
export { default as Commit } from './commit.model'
export { default as Repository } from './repository.model'
