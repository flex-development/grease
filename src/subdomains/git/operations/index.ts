/**
 * @file Entry Point - Operations
 * @module grease/git/operations
 */

export { default as TagOperationHandler } from './tag.handler'
export { default as TagOperation, type TagOperationDTO } from './tag.operation'
