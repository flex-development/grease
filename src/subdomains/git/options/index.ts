/**
 * @file Entry Point - Options
 * @module grease/git/options
 */

export { default as CommitGrammarOptions } from './commit-grammar.options'
export { default as GitCommitOptions } from './git-commit.options'
export { default as GitTagOptions } from './git-tag.options'
export { default as GitOptions } from './git.options'
