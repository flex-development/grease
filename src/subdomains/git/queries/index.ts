/**
 * @file Entry Point - Queries
 * @module grease/git/queries
 */

export { default as CommitQueryHandler } from './commit.handler'
export { default as CommitQuery } from './commit.query'
export { default as TagQueryHandler } from './tag.handler'
export { default as TagQuery } from './tag.query'
