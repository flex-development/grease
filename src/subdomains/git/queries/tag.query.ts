/**
 * @file Queries - TagQuery
 * @module grease/git/queries/TagQuery
 */

import { GitOptions } from '#src/git/options'
import { get, sift, type Partial } from '@flex-development/tutils'
import { IsArray, IsString, MinLength } from 'class-validator'

/**
 * Git tags query.
 *
 * @see {@linkcode GitOptions}
 * @see https://git-scm.com/docs/git-tag
 *
 * @class
 * @extends {GitOptions}
 */
class TagQuery extends GitOptions {
  /**
   * Sorting configuration.
   *
   * @see https://git-scm.com/docs/git-tag#Documentation/git-tag.txt---sortltkeygt
   *
   * @default ['-creatordate']
   *
   * @public
   * @instance
   * @member {string[]} to
   */
  @IsString({ each: true })
  @MinLength(1, { each: true })
  @IsArray()
  public sort: string[]

  /**
   * Create a new release tags query.
   *
   * @param {Partial<TagQuery>} [params] - Query parameters
   */
  constructor(params?: Partial<TagQuery>) {
    super(params)
    this.sort = sift(get(params, 'sort', ['-creatordate']))
  }
}

export default TagQuery
