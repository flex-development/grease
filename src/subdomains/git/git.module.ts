/**
 * @file GitModule
 * @module grease/git/GitModule
 */

import { LogModule } from '#src/log'
import { ValidationService } from '#src/providers'
import { Global, Module } from '@nestjs/common'
import { CqrsModule } from '@nestjs/cqrs'
import { TagOperationHandler } from './operations'
import { GitService } from './providers'
import { CommitQueryHandler, TagQueryHandler } from './queries'

/**
 * Git operations module.
 *
 * @class
 */
@Global()
@Module({
  exports: [GitService],
  imports: [CqrsModule, LogModule],
  providers: [
    CommitQueryHandler,
    GitService,
    TagOperationHandler,
    TagQueryHandler,
    ValidationService
  ]
})
class GitModule {}

export default GitModule
