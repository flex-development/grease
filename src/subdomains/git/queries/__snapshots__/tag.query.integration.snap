// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`integration:git/queries/TagQuery > validation > should fail validation if schema is invalid 1`] = `
[
  ValidationError {
    "children": [],
    "constraints": {
      "isBoolean": "color must be a boolean value",
    },
    "property": "color",
    "value": 3,
  },
  ValidationError {
    "children": [],
    "constraints": {
      "isFile": "config must be a file path or url",
    },
    "property": "config",
    "value": 0,
  },
  ValidationError {
    "children": [],
    "constraints": {
      "isEnum": "level must be one of the following values: debug, error, fatal, info, log, silent, trace, verbose, warn",
    },
    "property": "level",
    "value": "quiet",
  },
  ValidationError {
    "children": [],
    "constraints": {
      "isString": "tagprefix must be a string",
    },
    "property": "tagprefix",
    "value": /grease@/,
  },
  ValidationError {
    "children": [],
    "constraints": {
      "isBoolean": "unstable must be a boolean value",
    },
    "property": "unstable",
    "value": 1,
  },
]
`;
